# **Data Analaysis for *A Comparative Analysis of Content Across Major Streaming Services***

#### **Authors**: Karis Park and Elise Huhn

### **Research Questions:**

1. Are there significant differences in genre distribution across streaming platforms?
2. Is there a difference between the platforms in their balance between movies and television shows?
3. Can observations be accurately classified as "Movie" or "TV Show" by their known attributes?

## **Descriptive Statistics**

In [1]:
#Library Imports
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

In [2]:
#Read in the datasets
streaming_full = pd.read_csv("Data/streaming_combines.csv")

In [ ]:
#show overall numerical summaries
streaming_full.describe()

,release_year,action,comedy,adventure,animation,anime,drama,international,kids,horror
count,22998.000000,22998.000000,22998.000000,22998.000000,22998.000000,22998.000000,22998.000000,22998.000000,22998.000000,22998.000000
mean,2010.811244,0.140838,0.160492,0.080094,0.048961,0.028698,0.205583,0.203235,0.079963,0.070050
std,15.401142,0.347862,0.367070,0.271444,0.215791,0.166960,0.404136,0.402415,0.271243,0.255236
min,1920.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,2010.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,2016.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
75%,2019.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
max,2021.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000


**Interpretation:** These numerical summaries show that the oldest screen media in out dataset was released in 1920, whereas the most recent was released in 2021. The median release date is 2016 and the mean is 2010, which indicates that the majority of the motion pictures were released after the 2000s. The numerical summaries for the other quatitative variables are less useful in determining their distributions because they are binary variables. 

In [31]:
#Frequency of each Streaming Service
streaming_full["channel_streaming"].value_counts()

channel_streaming
amazon_prime    9668
netflix         8807
hulu            3073
disney_plus     1450
Name: count, dtype: int64

**Interpretation:** Amazon Prime is the streaming service with the most amount of observations (9668). The second most frequent is Netflix (8807), and the third is Hulu (3073). The streaming service with the least amount of observations is Disney Plus (1450), which has less than one-sixth the amount of observations as Amazon Prime. Thus, our data is not evantly distributed among the different streaming services. 

In [32]:
#Frequency of Movies and TV Shows
streaming_full["movie_or_serie"].value_counts()

movie_or_serie
Movie      16481
TV Show     6517
Name: count, dtype: int64

**Interpretation:** There are 2.5 times as many movies in the dataset as there are TV shows. So, our data is not evenly distributed among the two different types of motion pictures. 

In [28]:
#Total Counts of Each Genre
genres = ['action', 'comedy', 'adventure', 'animation','anime', 'drama', 'international', 'kids', 'horror']

print("Frequency of Each Genre:")
streaming_full[genres].sum().sort_values(ascending=False)

Frequency of Each Genre:


drama            4728
international    4674
comedy           3691
action           3239
adventure        1842
kids             1839
horror           1611
animation        1126
anime             660
dtype: int64

**Interpretation:** The genre with the most observations is drama. Anime is the genre with the least amount of observations. There are about 7 times as many drama observations as there are amine observations. There are a similar numer of drama and international observations. Similarly, there are a similar number of comedy and action observations as well as a similar numer of adventure, kids, horror, and animation movies. The genres are not evenly distributed throughout the dataset. 

In [21]:
#frequency of each genre for each streaming service
genres = ['action', 'comedy', 'adventure', 'animation','anime', 'drama', 'international', 'kids', 'horror']

genre_table = streaming_full.groupby("channel_streaming")[genres].sum()

print("Genre Frequency per Streaming Service:")
print(genre_table)

Genre Frequency per Streaming Service:
                   action  comedy  adventure  animation  anime  drama  \
channel_streaming                                                       
amazon_prime         1657    2099        259        547     80   3687   
disney_plus             0     526          0        542      4    134   
hulu                  555     667        556         37    329    907   
netflix              1027     399       1027          0    247      0   

                   international  kids  horror  
channel_streaming                               
amazon_prime                 355  1085     875  
disney_plus                    0   141       0  
hulu                         216   162     304  
netflix                     4103   451     432  


In [ ]:
#frequency of each genre for each streaming service
print("Movie or Streaming distributiton for Streaming Service:")
proportions_movie_show = (streaming_full.groupby("channel_streaming")["movie_or_serie"].value_counts(normalize=True).unstack(fill_value=0))
print(proportions_movie_show)

Movie or Streaming distributiton for Streaming Service:
movie_or_serie        Movie   TV Show
channel_streaming                    
amazon_prime       0.808233  0.191767
disney_plus        0.725517  0.274483
hulu               0.482916  0.517084
netflix            0.696151  0.303849


### **Documentation**

First, we documented the numerical summaries of all of the quantitative varibles. This was most useful for determining the distribution of the release years for the screen media. Since each genre column is bianary, we displayed the frequency of each genre across all streaming services to better determine the genre distribution. Then, we provided the frequencies for both the categorical variables of channel_streaming and movie_or_serie to show their distributions. 

These numerical representations allowed us to understand the distributions of out variables to begin investigating our research questions. Namely we focused on the distribution of genres and streaming services to investigate our first research question: "Are there significant differences in genre distribution across streaming platforms?" We also investigated the distributions of movies and TV shows for our second research question: "Is there a difference between the platforms in their balance between movies and television shows?" And finally the combination of all of the sumerical summaries will aid us in determining our thrid research question: "Can observations be accurately classified as "Movie" or "TV Show" by their known attributes?"

### **Reporting and Interpretation**

Interspresed throughout code whomp whomp

## **Inferential Statistics**

### **Documentation**

### **Reporting and Interpretation**

## **Graphical Analysis**

### **Documentation**

### **Reporting and Interpretation**

## **Comparative Analysis**

### **Documentation**

### **Reporting and Interpretation**

## **Multivariate Analysis**

### **Documentation**

### **Reporting and Interpretation**

## **Synthesis**

### **Documentation**

### **Reporting and Interpretation**

## **Reflection**

### **Documentation**

### **Reporting and Interpretation**